backend/packages/Upgrade/src/api/services/StratificationService.ts

```diff
@@ -0,0 +1,225 @@
+import { Service } from 'typedi';
+import { OrmRepository } from 'typeorm-typedi-extensions';
+import { UpgradeLogger } from '../../lib/logger/UpgradeLogger';
+import { SERVER_ERROR } from 'upgrade_types';
+import { In, getConnection } from 'typeorm';
+import Papa from 'papaparse';
+import { FactorStrata, StratificationInputValidator } from '../controllers/validators/StratificationValidator';
+import { ExperimentUser } from '../models/ExperimentUser';
+import { StratificationFactor } from '../models/StratificationFactor';
+import { UserStratificationFactor } from '../models/UserStratificationFactor';
+import { StratificationFactorRepository } from '../repositories/StratificationFactorRepository';
+import { ErrorWithType } from '../errors/ErrorWithType';
+@Service()
+export class StratificationService {
+  constructor(
+    @OrmRepository()
+    private stratificationFactorRepository: StratificationFactorRepository
+  ) {}
+
+  private calculateStratificationResult(
+    results: { factor: string; value: string; count: string; experimentIds: string[] }[]
+  ): FactorStrata[] {
+    const formattedResults = results.reduce((formatted, result) => {
+      const { factor, value, count, experimentIds } = result;
+      if (!formatted[factor]) {
+        formatted[factor] = { factor, factorValue: {}, experimentIds };
+      }
+      if (value !== 'N/A') {
```

Comment:
will this 'N/A' string for sure come this way? and never like `n/a` or `n / a` or whatever?

Reply:
Good point about this string. I'd suggest if we make a default string value it should be just `NA`. We need to make sure this is described in the import UI and handled well for MVP.

Reply:
This `if` condition should be removed. It's not used now.